src/cript/nodes/supporting_nodes/group.py

```diff
@@ -10,6 +16,196 @@ class Group(BaseNode):
+        name: str = ""
+        admins: List[Any] = None
+        users: List[Any] = None
+        notes: str = ""
+
+    _json_attrs: JsonAttributes = JsonAttributes()
+
+    def __init__(self, name: str, admins: List[Any], users: List[Any] = None, **kwargs):
+        """
+        constructor for a Group node
+
+        Parameters
+        ----------
+        name: str
+            group name
+        admins: Any
+            administrator of this group
+        users: List[Any]
+            list of users that are in this Group
+
+        Returns
+        -------
+        None
+        """
+        super().__init__(node="Group")
+        self._json_attrs = replace(self._json_attrs, name=name, admins=admins, users=users)
+        self.validate()
+
+    def validate(self) -> None:
+        """
+        validates Group node
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        CRIPTNodeSchemaError
+        """
         pass
 
-    pass
+    # ------------------ Properties ------------------
+
+    # Group name
+    @property
+    def name(self) -> str:
+        """
+        name property getter method
+
+        Returns
+        -------
+        name: str
+            group name
+        """
+        return self._json_attrs.name
+
+    @name.setter
+    def name(self, new_name: str) -> None:
+        """
+        Setter for the group name
+
+        User cannot set the name for any nodes.
+        Attempt to do so will raise an UneditableAttributeError
+
+        Parameters
+        ----------
+        new_name: str
+            new group name
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        UneditableAttributeError
+            If user attempts to set this property
+        """
+        raise UneditableAttributeError
+
+    # admins
+    @property
+    def admins(self) -> List[Any]:
+        """
+        name property getter method
+
+        Returns
+        -------
+        admins: List[Any]
+            an admin or list of admins
+        """
+        return self._json_attrs.admins
+
+    @admins.setter
+    def admins(self, new_admins: List[Any]) -> None:
+        """
+        sets admins for the Group node
+
+        User cannot set the admins of the group node.
+        Attempt to do so will raise an UneditableAttributeError
+
+        Parameters
+        ----------
+        new_admins
+
+        Returns
+        -------
+        None
+
+        Raises
+        ------
+        UneditableAttributeError
+            If user attempts to set this property
+        """
+        raise UneditableAttributeError
+
+    @property
+    def users(self) -> List[Any]:
+        """
+        users that belong to this group
+
+        Returns
+        -------
+        List[Any]
+            list of users that belong to this group
+        """
+        return self._json_attrs.users
+
+    @users.setter
+    def users(self, new_users: List[Any]) -> None:
```

Comment:
I am a big advocate for accurate, understandable, and timely user feedback.

I updated the code and removed all the setters to for `Group` and `User` nodes for now. This will trigger an `AttributeError` if the user attempts to set it. For now, I think this is good enough, so that we can play around with it and debug it, but in future updates we probably want to come back and update all read-only attributes to be more user friendly.

I wrote an [issue](https://github.com/C-Accel-CRIPT/Python-SDK/issues/45) for it so we do not forget about it.